code/game/objects/structures/mineral_doors.dm

```diff
@@ -151,18 +157,31 @@
 	sheetType = /obj/item/stack/sheet/mineral/silver
 	max_integrity = 300
 
+/obj/structure/mineral_door/silver/Initialize()
+	. = ..()
+	AddComponent(/datum/component/rad_insulation, RAD_HEAVY_INSULATION)
+
 /obj/structure/mineral_door/gold
 	name = "gold door"
 	icon_state = "gold"
 	sheetType = /obj/item/stack/sheet/mineral/gold
 
+/obj/structure/mineral_door/gold/Initialize()
+	. = ..()
+	AddComponent(/datum/component/rad_insulation, RAD_HEAVY_INSULATION)
+
 /obj/structure/mineral_door/uranium
 	name = "uranium door"
 	icon_state = "uranium"
 	sheetType = /obj/item/stack/sheet/mineral/uranium
 	max_integrity = 300
 	light_range = 2
 
+/obj/structure/mineral_door/uranium/Initialize()
+	. = ..()
+	var/datum/component/kill = GetComponent(/datum/component/rad_insulation)
+	kill.Destroy()

```

Comment:
wat

Reply:
sir, you know qdel exists right?

Reply:
...right, yes. I was rushing this part and was going off of component documentation looking for something like RemoveComponent() and it said destroy was public so I used it >.>